### ISSUE TITLE
Should this template repo dependency pin (or use a different renovate config)?

### ISSUE THREAD

**Issue by johnslemmer** [t3-oss/create-t3-turbo#214]
I'm curious if the [renovate config](https://github.com/t3-oss/create-t3-turbo/blob/6c35df5ac1805d950f740b7414ee6b9b15bd54d6/renovate.json) should change at all or if this repo should adopt dependency pinning. And I wondered what the T3 community thought were the best practices.

As it stands today with the current `package.json` files and Renovate Bot config any dependencies that include a range will not automatically generate a new renovate PR (if a new published version is within range).  Say, for example `eslint` publishes a new minor version (which they have, 8.36.0).  Renovate will not (and has not) generated a PR to update this dependency... and it is within the acceptable range.  For this repo, and any repo based on it, you would have to manually `pnpm update eslint` to upgrade eslint.

I had started my own project based on this repo (which is amazing by the way, thanks all) and was confused why Renovate Bot wasn't bugging me more.  So I did some reading:

https://docs.renovatebot.com/dependency-pinning/
https://docs.renovatebot.com/configuration-options/#rangestrategy

tldr; renovate recommends pinning all dependencies in `package.json` files to specific versions for web apps (and I would guess react native apps too).  But for libraries that get published and consumed they should use ranges.  However, pinning dependencies comes at the cost of renovate bugging you a lot more.

There are a few options I think for this repo.  And curious to get the communities thoughts.

1. Leave the `package.json` files as they are (with ranges), update the renovate config to `rangeStrategy` to `update-lockfile`, but remove the lock file from this template repo. This means that this repo would get bugged by renovate the same amount.  But people who use this template, on first `pnpm i` will generate a lock file and download the most in range dependencies.  And then renovate will start bugging them about updating dependencies that have been published (even in range ones).
1. Go ahead and pin all the dependencies in this repo.  Based on industry best practice, this seems ideal.  Every dependency update will be its own PR.  Which is noisy, but when you are trying to ship a stable product having each change run through CI makes it easier to see what will break things, and what won't.  But this repo will get a lot more PRs from renovate.  Which sucks for maintainers.
1. Do something more hybrid.  The expo and nextjs apps can pin their dependencies.  But the shared packages can stay as is, as they technically might be consumed outside this repo.  In this case the renovate config I believe can stay as is.
1. Don't change anything and just document about renovate in the  README a bit more.

What do you all think?  Any other ideas?

**Comment by juliusmarminge** [comment]
Interesting... I'd actually expect Renovate to update deps even when they are within the accepted range :/

I don't care about the noise myself - but I initially added Renovate just so that the repo would stay up to update for people to use an up-to-date template. I don't fully expect people to keep using Renovate after they have initialiazed the project?

I don't want to remove the lockfile though. Ideally Renovate would just update deps even with an "open range" if that makes sense 🤷🏼‍♂️

**Comment by johnslemmer** [comment]
@juliusmarminge I just verified on my repo.  I think the renovate config for `rangeStrategy` you are looking for is `bump`.  This will update things in range.  As well as update the range in the `package.json` file to have this new version as the minimum.  You can keep the lock file.  But you will get more noise.  But things will stay more up to date.  Happy to make a PR if you want.

**Comment by juliusmarminge** [comment]
I'm fine with noise. Feel free to PR that change :)
